root/static/scripts/common/components/FilterForm.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import SelectField from '../../../../components/SelectField';
+
+export type FilterFormT =
+  & FormT<{|
+      +artist_credit_id: ReadOnlyFieldT<number>,
+      +name: ReadOnlyFieldT<string>,
+      +type_id?: ReadOnlyFieldT<number>,
+    |}>
+  & {
+      +entity_type: 'recording' | 'release' | 'release_group',
+      +options_artist_credit_id: SelectOptionsT,
+      +options_type_id?: SelectOptionsT,
+    };
+
+type Props = {|
+  +form: FilterFormT,
+|};
+
+function getSubmitText(type: string) {
+  switch (type) {
+    case 'recording':
+      return l('Filter recordings');
+    case 'release':
+      return l('Filter releases');
+    case 'release_group':
+      return l('Filter release groups');
+  }
+  return '';
+}
+
+const FilterForm = ({form}: Props) => {
+  const typeIdField = form.field.type_id;
+  const typeIdOptions = form.options_type_id;
+  const artistCreditIdField = form.field.artist_credit_id;
+  const artistCreditIdOptions = form.options_artist_credit_id;
+
+  return (
+    <div id="filter">
+      <form method="get">
+        <table>
+          <tbody>
+            {typeIdField && typeIdOptions ? (
+              <tr>
+                <td style={{whiteSpace: 'nowrap'}}>{l('Type:')}</td>
+                <td>
+                  <SelectField
+                    field={typeIdField}
+                    options={{grouped: false, options: typeIdOptions}}
+                    style={{maxWidth: '40em'}}
+                    uncontrolled
+                  />
+                </td>
+              </tr>
+            ) : null}
+
+            {artistCreditIdField && artistCreditIdOptions ? (
+              <tr>
+                <td style={{whiteSpace: 'nowrap'}}>
+                  {l('Artist credit:')}
+                </td>
+                <td>
+                  <SelectField
+                    field={artistCreditIdField}
+                    options={{
+                      grouped: false,
+                      options: artistCreditIdOptions,
+                    }}
+                    style={{maxWidth: '40em'}}
+                    uncontrolled
+                  />
+                </td>
+              </tr>
+            ) : null}
+
+            <tr>
+              <td>{l('Name:')}</td>
+              <td>
+                <input
+                  name={form.field.name.html_name}
+                  size="47"
+                  type="text"
+                  value={form.field.name.value}
```

Comment:
On trying to change a name after submitting:

`Warning: Failed prop type: You provided a `value` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultValue`. Otherwise, set either `onChange` or `readOnly`.`